bundle/edu.gemini.seqexec.server/src/main/scala/edu/gemini/seqexec/server/Executor.scala

```diff
@@ -0,0 +1,49 @@
+package edu.gemini.seqexec.server
+
+import edu.gemini.seqexec.server.SeqexecFailure._
+import edu.gemini.spModel.config2.{Config, ConfigSequence, ItemKey}
+import edu.gemini.spModel.obscomp.InstConstants.INSTRUMENT_NAME_PROP
+import edu.gemini.spModel.seqcomp.SeqConfigNames.INSTRUMENT_KEY
+
+import scalaz._
+import Scalaz._
+
+import scalaz.concurrent.Task
+
+/**
+ * Created by jluhrs on 4/28/15.
+ */
+final class Executor(sequenceConfig: ConfigSequence) {
+
+  def run: \/[Throwable, List[StepResult]] = runSequence(sequenceConfig.getAllSteps.toList).attemptRun
+
+  private def runSequence(config: List[Config]): Task[List[StepResult]] = {
+    (config map (runStep(_))).sequenceU
+  }
+
+  private def runStep(stepConfig: Config): Task[StepResult] = {
+    val instName = stepConfig.getItemValue(new ItemKey(INSTRUMENT_KEY, INSTRUMENT_NAME_PROP))
+    val instrument = instName match {
+      case GMOS_S.name => Some(GMOS_S)
+      case _           => None
+    }
+
+    instrument map (a => {
+      val systems: List[System] = List(TCS, a)
+      for {
+        r <- Task.gatherUnordered( systems map (x => x.configure(stepConfig)) , true)
+        s <- a.observe(stepConfig)
+      } yield StepResult(r, s)
+    }) getOrElse Task.fail(UnrecognizedInstrument(instName.toString))
```

Comment:
Agree, let's get rid of exceptions. With `EitherT` you can use `.sequence` to go from `List[EitherT[SeqFailure,A]]` to `Task[EitherT[SeqFailure, List[A]]]` which I believe is what you want.